### PR TITLE
Fix/uc1 test

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

This is a assorted fixed related to UC1 regression testing. 
- Mobility path and operation should not be published if the vehicle is not engaged  for uc1.
- bsm id will generate random id from the beginning and uses fixed id if it random_generation is toggled false
- Mobility path's timestamp should be still up to date when waiting on a stop line

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1926 BSM ID
https://github.com/usdot-fhwa-stol/carma-platform/issues/1937 Mobility Path timestamp
https://github.com/usdot-fhwa-stol/carma-platform/issues/1938 Mobility Path & Operation should not publish when not engaged

## Motivation and Context
Use case 1 regression testing fixes.

## How Has This Been Tested?
Integration tested on blue lexus and white pacifica 

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: sci_strategic_plugin/config/parameters.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ vehicle_decel_limit_multiplier : 0.65
 vehicle_accel_limit_multiplier : 0.65
 
 # Double: A buffer before of the stopping location which will still be considered a valid stop. Units in meters
-stop_line_buffer : 3.5
+stop_line_buffer : 10
```
Comment (msmcconnell) [inline sci_strategic_plugin/config/parameters.yaml:8]: Is this acceptable? How does it compare with UC2?

File: stop_and_wait_plugin/config/parameters.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ centerline_sampling_spacing: 1.0
 # Double: The default buffer in meters used for where the vehicle can come to a stop during execution. 
 #         This value is overriden by the first value in maneuver.parameters.float_valued_meta_data[]
 # Units: m
-default_stopping_buffer: 5.0
+default_stopping_buffer: 15.0
```
Comment (msmcconnell) [inline stop_and_wait_plugin/config/parameters.yaml:29]: Same question on acceptability as above.

Comment (MishkaMN) [reply stop_and_wait_plugin/config/parameters.yaml:29]: This parameter is being overridden, so we can actually keep the older value. So I reversed it.

File: sci_strategic_plugin/config/parameters.yaml (hunk 1 above)
Comment (MishkaMN) [reply sci_strategic_plugin/config/parameters.yaml:8]: UC2 has stopping location buffer of 8 meters, which I think is comparable to this value. UC2 has a little more sophisticated guard for stopping case that allows it to stop closer (once it is decided to stop, uc2 plugin will keep it slowing, while uc1 may try to lane follow again in stopping distance). So another way to make it stop more consistently without changing the code was increasing this value.

**Pull request merged**